### DISCUSSION TITLE
Static backend method names must be unique accross all controllers, or URLs clash?

### DISCUSSION THREAD

**Discussion by SpeedoPasanen** [noam-honig/radweb#189] [Q&A] [answered]
Hi,

exploring Remult for the first time, and it seems awesome, but I'm stuck with this:

Static backend methods are always routed to `/api/methodName` and I can't find a way to change that. Is there? Eg. I have `AController` and `BController`, both have a static method `hello`. When I call hello of either controller in the frontend, requests go to `/api/hello` and call the method in AController. I would like URLs something like `/api/[optional_global_prefix_for_static_medhods]/ControllerA/hello`. 

I tried adding the `Controller` decorator but it's `type` argument has no effect on this.

**Comment by noam-honig** [comment, marked as answer]
Hi @SpeedoPasanen,
In the latest release [v0.20.2](https://github.com/remult/remult/releases/tag/v0.20.2)
We've added an option for the backend method called `apiPrefix` to help you achieve what you've wanted.

```ts
class myClass{
  @BackendMethod({allowed:true, apiPrefix:'prefixa/prefixb'})
  static async methodName(){
    return "works"
  }
}
```
Will result in an api:
`/api/prefixa/prefixb/methodName`

This will help you organize your backend methods api

Can you take the latest version, try it out and let me know if it works for you?

**Reply by SpeedoPasanen** [reply]
It works perfectly, thank you ♥♥♥

**Reply by aepshtin** [reply]
@noam-honig 
My suggestion is to take the name of the class or the file as part of the path.
like: /api/myClass/methodName in the above example.
What do you think?

**Reply by noam-honig** [reply]
Hi @aepshtin - that is not possible because many frontend bundlers change the class name as they create the bundle, and this will cause a mismatch between for front and back
